src/data/finset/lattice.lean

```diff
@@ -142,6 +142,14 @@ begin
     exact ⟨le_trans hay hyz, le_trans hsx_sup hxz⟩, },
 end
 
+-- If we acquire sublattices
+-- the hypotheses should be reformulated as `s : subsemilattice_sup_bot`
+lemma sup_mem
+  (s : set α) (w₁ : ⊥ ∈ s) (w₂ : ∀ x y ∈ s, x ⊔ y ∈ s)
+  {ι : Type*} (t : finset ι) (p : ι → α) (h : ∀ i ∈ t, p i ∈ s) :
+  t.sup p ∈ s :=
+@sup_induction _ _ _ _ _ (∈ s) w₁ w₂ h
```

Comment:
This proof also works:
```suggestion
(sup_induction w₁ w₂ h : id (∈ s) _)
```
there's something very odd going on with the elaborator here

Reply:
I'm not sure that's any clearer: I'm struggling to understand what the `: id (∈ s) _` is doing!

Reply:
This works too:
```suggestion
(sup_induction w₁ w₂ h : ((∈ s) : _) _)
```

Reply:
As does
```suggestion
(sup_induction w₁ w₂ h : (λ x, x ∈ s) _)
```
I made [a Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Applying.20.28.CE.BB.20x.2C.20x.20.2B.20b.29.20and.20.28.2B.20b.29.20give.20different.20expressions/near/235072004)